src/server/debug/server/server.go

```diff
@@ -399,32 +408,25 @@ func (s *debugServer) makeLokiTask(app *debug.App, rp incProgressFunc) taskFunc
 	}
 }
 
-func (s *debugServer) makeProfilesTask(server debug.Debug_DumpV2Server, apps []*debug.App) taskFunc {
+func (s *debugServer) makeProfileTask(server debug.Debug_DumpV2Server, app *debug.App, pod *debug.Pod, rp incProgressFunc) taskFunc {
 	return func(ctx context.Context, dfs DumpFS) error {
+		defer rp(ctx)
+		if app.Timeout != nil && len(app.Pods) > 0 {
```

Comment:
I think we should do this timeout adjustment a level up.  Do we want to divide the per-Pod time by # containers * 2 here though?